backend/src/transactions/auth.go

```diff
@@ -0,0 +1,89 @@
+package transactions
+
+import (
+	"time"
+
+	"github.com/GenerateNU/sac/backend/src/errors"
+	"github.com/GenerateNU/sac/backend/src/models"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+func SaveOTP(db *gorm.DB, userID uuid.UUID, otp string) *errors.Error {
+	otpModel := models.Verification{
+		UserID:    userID,
+		Token:     otp,
+		ExpiresAt: time.Now().Add(time.Minute * 30).UTC(),
+		Type:      models.EmailVerificationType,
+	}
+
+	if err := db.Create(&otpModel).Error; err != nil {
+		return &errors.FailedToSaveOTP
+	}
+
+	return nil
+}
+
+func GetOTP(db *gorm.DB, userID uuid.UUID) (*models.Verification, *errors.Error) {
+	var otp models.Verification
+	if err := db.Where("user_id = ? AND type = ?", userID, models.EmailVerificationType).First(&otp).Error; err != nil {
+		return nil, &errors.FailedToGetOTP
+	}
+
+	return &otp, nil
+}
+
+func DeleteOTP(db *gorm.DB, userID uuid.UUID) *errors.Error {
```

Comment:
is this ever used? i couldn't find any usages.

what is the use case for this? (hinting that this likely shouldn't a blocking call at the end of a request but rather be executed by a background job)

Reply:
It is used for deleting OTP, I think I may have forgot to use it.

It's extremely important to delete the OTP and password reset token from the db, this is stated by password reset resources (although the OTP is less important as it gets invalidated by use, the password reset token doesn't and also just have it unnecessarily laying around has no use removing it when used is much better